File: async-scala-client/src/main/scala/io/swagger/client/api/AccountApi.scala
package io.swagger.client.api

import io.swagger.client.model.AccountCompany
import io.swagger.client.model.Body
import io.swagger.client.model.Inline_response_201
import io.swagger.client.model.Message
import java.util.UUID
import io.swagger.client.model.ValidationError
import com.wordnik.swagger.client._
import scala.concurrent.Future
import collection.mutable

class AccountApi(client: TransportClient, config: SwaggerConfig) extends ApiClient(client, config) {

  def accountsAccountIdCompaniesGet(accountId: UUID)(implicit reader: ClientResponseReader[List[AccountCompany]]): Future[List[AccountCompany]] = {
    // create path and map variables
    val path = (addFmt("/accounts/{accountId}/companies")
        replaceAll ("\\{" + "accountId" + "\\}",accountId.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (accountId == null) throw new Exception("Missing required parameter 'accountId' when calling AccountApi->accountsAccountIdCompaniesGet")


    val resFuture = client.submit("GET", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def createAccountCompany(accountId: UUID,
      body: Body)(implicit reader: ClientResponseReader[Inline_response_201], writer: RequestWriter[Body]): Future[Inline_response_201] = {
    // create path and map variables
    val path = (addFmt("/accounts/{accountId}/companies")
        replaceAll ("\\{" + "accountId" + "\\}",accountId.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (accountId == null) throw new Exception("Missing required parameter 'accountId' when calling AccountApi->createAccountCompany")

    if (body == null) throw new Exception("Missing required parameter 'body' when calling AccountApi->createAccountCompany")

    val resFuture = client.submit("POST", path, queryParams.toMap, headerParams.toMap, writer.write(body))
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }


}

File: CsharpDotNet2-client/src/main/CsharpDotNet2/IO/Swagger/Api/AccountApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IAccountApi
    {
        /// <summary>
        /// Retrieve Companies by account Retrieve list companies of one account
        /// </summary>
        /// <param name="accountId">Account ID</param>
        /// <returns>List&lt;AccountCompany&gt;</returns>
        List<AccountCompany> AccountsAccountIdCompaniesGet (Guid? accountId);
        /// <summary>
        /// Create company and your data configuration This method operation create a company for a account 
        /// </summary>
        /// <param name="accountId">Account ID</param>
        /// <param name="body">Company Code</param>
        /// <returns>InlineResponse201</returns>
        InlineResponse201 CreateAccountCompany (Guid? accountId, Body body);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class AccountApi : IAccountApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="AccountApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public AccountApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="AccountApi"/> class.
        /// </summary>
        /// <returns></returns>
        public AccountApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// Retrieve Companies by account Retrieve list companies of one account
        /// </summary>
        /// <param name="accountId">Account ID</param> 
        /// <returns>List&lt;AccountCompany&gt;</returns>            
        public List<AccountCompany> AccountsAccountIdCompaniesGet (Guid? accountId)
        {
            
            // verify the required parameter 'accountId' is set
            if (accountId == null) throw new ApiException(400, "Missing required parameter 'accountId' when calling AccountsAccountIdCompaniesGet");
            
    
            var path = "/accounts/{accountId}/companies";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "accountId" + "}", ApiClient.ParameterToString(accountId));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling AccountsAccountIdCompaniesGet: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling AccountsAccountIdCompaniesGet: " + response.ErrorMessage, response.ErrorMessage);
    
            return (List<AccountCompany>) ApiClient.Deserialize(response.Content, typeof(List<AccountCompany>), response.Headers);
        }
    
        /// <summary>
        /// Create company and your data configuration This method operation create a company for a account 
        /// </summary>
        /// <param name="accountId">Account ID</param> 
        /// <param name="body">Company Code</param> 
        /// <returns>InlineResponse201</returns>            
        public InlineResponse201 CreateAccountCompany (Guid? accountId, Body body)
        {
            
            // verify the required parameter 'accountId' is set
            if (accountId == null) throw new ApiException(400, "Missing required parameter 'accountId' when calling CreateAccountCompany");
            
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling CreateAccountCompany");
            
    
            var path = "/accounts/{accountId}/companies";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "accountId" + "}", ApiClient.ParameterToString(accountId));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                postBody = ApiClient.Serialize(body); // http body (model) parameter
    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateAccountCompany: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateAccountCompany: " + response.ErrorMessage, response.ErrorMessage);
    
            return (InlineResponse201) ApiClient.Deserialize(response.Content, typeof(InlineResponse201), response.Headers);
        }
    
    }
}
